### PR TITLE
Image uploading

### PR DESCRIPTION
**Pull request by thien0411**

### CODE REVIEW

File: FriendlyMenu/ViewModels/DishSummaryVM.cs [hunk 1]
```diff
@@ -6,10 +6,21 @@ namespace ViewModels
 {
     public class DishSummaryVM
     {
-        public string Name { get; set; }
+        public int Id { get; set; }
```
Comment (netechteam) [inline FriendlyMenu/ViewModels/DishSummaryVM.cs]: Change this to DishId

File: FriendlyMenu/ViewModels/DishSummaryVM.cs [hunk 2]
```diff
@@ -6,10 +6,21 @@ namespace ViewModels
 {
     public class DishSummaryVM
     {
-        public string Name { get; set; }
```
Comment (netechteam) [inline FriendlyMenu/ViewModels/DishSummaryVM.cs:9]: Do we need this?

File: FriendlyMenu/ViewModels/DishSummaryVM.cs [hunk 3]
```diff
@@ -6,10 +6,21 @@ namespace ViewModels
...
+        public int Id { get; set; }
+        public int RestaurantId { get; set; }
+        public int CategoryId { get; set; }
+        public string DishName { get; set; }
         public string Description { get; set; }
-        public string Price { get; set; }
+        public double PriceBreakfast { get; set; }
```
Comment (netechteam) [inline FriendlyMenu/ViewModels/DishSummaryVM.cs]: Change this to BreakfastPrice (same for lunch, combo, and dinner)

File: FriendlyMenu/DataAccessors/SharedHourDataAccessor.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+using System.Text;
+using DataModels;
+using System.Threading.Tasks;
+using Entities;
+using Microsoft.EntityFrameworkCore;
+using DbModels;
+
+namespace DataAccessors
+{
+    public class SharedHourDataAccessor : ISharedHourDataAccessor
```
Comment (netechteam) [inline FriendlyMenu/DataAccessors/SharedHourDataAccessor.cs]: Why do we need this?

File: FriendlyMenu/Interfaces/DataAccessors/ISharedHourDataAccessor.cs [hunk 5]
```diff
@@ -0,0 +1,13 @@
+﻿using DataModels;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Interfaces.DataAccessors
+{
+    public interface ISharedHourDataAccessor
```
Comment (netechteam) [inline FriendlyMenu/Interfaces/DataAccessors/ISharedHourDataAccessor.cs:9]: Why do we need this?

File: FriendlyMenu/DataAccessors/RestaurantHourDataAccessor.cs [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+using System.Collections.Generic;
+using System.Text;
+using DataModels;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using DbModels;
+
+namespace DataAccessors
+{
+    class RestaurantHourDataAccessor : IRestaurantHourDataAccessor
```
Comment (netechteam) [inline FriendlyMenu/DataAccessors/RestaurantHourDataAccessor.cs]: Why do we need this?

File: FriendlyMenu/Interfaces/DataAccessors/IRestaurantHourDataAccessor.cs [hunk 7]
```diff
@@ -0,0 +1,13 @@
+﻿using DataModels;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Interfaces.DataAccessors
+{
+    public interface IRestaurantHourDataAccessor
```
Comment (netechteam) [inline FriendlyMenu/Interfaces/DataAccessors/IRestaurantHourDataAccessor.cs:9]: Why do we need this?

File: FriendlyMenu/Interfaces/DataAccessors/IRestaurantContactDataAccessor.cs [hunk 8]
```diff
@@ -0,0 +1,13 @@
+﻿using DataModels;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Interfaces.DataAccessors
+{
+    public interface IRestaurantContactDataAccessor
```
Comment (netechteam) [inline FriendlyMenu/Interfaces/DataAccessors/IRestaurantContactDataAccessor.cs:9]: Why do we need this?

File: FriendlyMenu/Interfaces/DataAccessors/IIngredientTypeDataAccessor.cs [hunk 9]
```diff
@@ -0,0 +1,13 @@
+﻿using DataModels;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Interfaces.DataAccessors
+{
+    public interface IIngredientTypeDataAccessor
```
Comment (netechteam) [inline FriendlyMenu/Interfaces/DataAccessors/IIngredientTypeDataAccessor.cs:9]: Why do we need this?

File: FriendlyMenu/Interfaces/DataAccessors/IIngredientDataAccessor.cs [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Interfaces.DataAccessors
+{
+    public interface IIngredientDataAccessor
+    {
+        Task<IngredientDM> GetIngredient(int ingredientID);
```
Comment (netechteam) [inline FriendlyMenu/Interfaces/DataAccessors/IIngredientDataAccessor.cs:11]: Why do we need this?

File: FriendlyMenu/Entities/DatabaseContext.cs [hunk 11]
```diff
@@ -10,12 +10,14 @@ public class DatabaseContext : DbContext
         public DbSet<tblRestaurant> Restaurant { get; set; }
         public DbSet<tblAddress> Address { get; set; }
         public DbSet<tblDish> Dish { get; set; }
-        public DbSet<tblDishIngredient> DishIngredient { get; set; }
+        public DbSet<tblDishIngredients> DishIngredients { get; set; }
```
Comment (netechteam) [inline FriendlyMenu/Entities/DatabaseContext.cs:13]: Why is this plural?

File: FriendlyMenu/DataAccessors/RestaurantDataAccessor.cs [hunk 12]
```diff
@@ -17,23 +19,78 @@ public RestaurantDataAccessor(DatabaseContext databaseContext)
...
-                Id = restaurant.Id,
-                RestaurantName = restaurant.RestaurantName
+                CategoryName = categoryName,
+                DishSummaryDM = allDBDishes
+
             };
+            return categoryDishes;
+        }
+
+        public void GetDishesByCategoryId(int categoryId)
```
Comment (netechteam) [inline FriendlyMenu/DataAccessors/RestaurantDataAccessor.cs]: Remove this

File: FriendlyMenu/DataAccessors/RestaurantDataAccessor.cs [hunk 13]
```diff
@@ -17,23 +19,78 @@ public RestaurantDataAccessor(DatabaseContext databaseContext)
...
+        {
+
         }
+
+        public void GetDishById(int categoryId)
```
Comment (netechteam) [inline FriendlyMenu/DataAccessors/RestaurantDataAccessor.cs]: And this

File: FriendlyMenu/DataAccessors/RestaurantDataAccessor.cs [hunk 14]
```diff
@@ -17,23 +19,78 @@ public RestaurantDataAccessor(DatabaseContext databaseContext)
...
-            return BuildRestaurantDM(restaurant);
-        }
 
-        private RestaurantDM BuildRestaurantDM(tblRestaurant restaurant)
+
+        public async Task<CategoryDishesDM> GetDishesByCategory(int categoryId, int restaurantId)
         {
-            if (restaurant == null)
-                return null;
+            var allDBDishes = _databaseContext.Dish.Where(x => x.RestaurantId == 1 && x.CategoryId == categoryId).Select(y => new DishSummaryDM
```
Comment (netechteam) [inline FriendlyMenu/DataAccessors/RestaurantDataAccessor.cs]: RestaurantId == restaurantId

**Review by netechteam — CHANGES_REQUESTED** [review]
Please look at the comments

File: FriendlyMenu/DataAccessors/RestaurantHourDataAccessor.cs (hunk 6 above)
Comment (thien0411) [reply FriendlyMenu/DataAccessors/RestaurantHourDataAccessor.cs]: all the data accessors other than RestaurantDataAccessor arent not include in the project. thats the old code. I can Delete it now.

File: FriendlyMenu/ViewModels/DishSummaryVM.cs (hunk 2 above)
Comment (thien0411) [reply FriendlyMenu/ViewModels/DishSummaryVM.cs:9]: we have an DishName Property, so we delete it

**Pull request merged**